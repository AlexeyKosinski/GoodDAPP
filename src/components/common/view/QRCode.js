// @flow
import React from 'react'
import QRCodeReact from 'qrcode.react'
import { View } from 'react-native'
import normalize from 'react-native-elements/src/helpers/normalizeText'
import { withStyles } from '../../../lib/styles'

const QRCode = ({ styles, ...props }: any) => {
  return (
    <View style={styles.qrWrapper}>
      <View style={styles.qrCode}>
        <QRCodeReact {...props} />
      </View>
    </View>
  )
}

const getStylesFromProps = ({ theme }) => {
  return {
    qrCode: {
      padding: normalize(16),
      borderColor: theme.colors.primary,
      borderWidth: 1,
      borderRadius: normalize(5),
    },
    qrWrapper: {
      justifyContent: 'center',
      alignItems: 'center',
    },
  }
}

export default withStyles(getStylesFromProps)(QRCode)
