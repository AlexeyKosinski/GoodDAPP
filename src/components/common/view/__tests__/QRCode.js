import React from 'react'
import renderer from 'react-test-renderer'
import QRCode from '../QRCode'

// Note: test renderer must be required after react-native.

describe('QRCode', () => {
  it('renders without errors', () => {
    const tree = renderer.create(<QRCode value={'code'} />)
    expect(tree.toJSON()).toBeTruthy()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<QRCode value={'code'} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('matches snapshot', () => {
    const component = renderer.create(<QRCode value={''} />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
