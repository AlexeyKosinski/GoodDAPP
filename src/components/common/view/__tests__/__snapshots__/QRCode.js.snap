// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QRCode matches snapshot 1`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxPack": "center",
      "WebkitJustifyContent": "center",
      "alignItems": "center",
      "justifyContent": "center",
      "msFlexAlign": "center",
      "msFlexPack": "center",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomColor": "rgba(98,0,238,1.00)",
        "borderBottomLeftRadius": "5px",
        "borderBottomRightRadius": "5px",
        "borderBottomWidth": "1px",
        "borderLeftColor": "rgba(98,0,238,1.00)",
        "borderLeftWidth": "1px",
        "borderRightColor": "rgba(98,0,238,1.00)",
        "borderRightWidth": "1px",
        "borderTopColor": "rgba(98,0,238,1.00)",
        "borderTopLeftRadius": "5px",
        "borderTopRightRadius": "5px",
        "borderTopWidth": "1px",
        "paddingBottom": "16px",
        "paddingLeft": "16px",
        "paddingRight": "16px",
        "paddingTop": "16px",
      }
    }
  >
    <canvas
      height={128}
      style={
        Object {
          "height": 128,
          "width": 128,
        }
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "backdrop": "rgba(0, 0, 0, 0.5)",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "notification": "#f50057",
            "placeholder": "rgba(0, 0, 0, 0.54)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "Roboto-Light, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "medium": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "regular": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "thin": "Roboto-Thin, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          },
          "roundness": 4,
        }
      }
      width={128}
    />
  </div>
</div>
`;

exports[`QRCode matches snapshot 2`] = `
<div
  className="css-view-1dbjc4n"
  style={
    Object {
      "WebkitAlignItems": "center",
      "WebkitBoxAlign": "center",
      "WebkitBoxPack": "center",
      "WebkitJustifyContent": "center",
      "alignItems": "center",
      "justifyContent": "center",
      "msFlexAlign": "center",
      "msFlexPack": "center",
    }
  }
>
  <div
    className="css-view-1dbjc4n"
    style={
      Object {
        "borderBottomColor": "rgba(98,0,238,1.00)",
        "borderBottomLeftRadius": "5px",
        "borderBottomRightRadius": "5px",
        "borderBottomWidth": "1px",
        "borderLeftColor": "rgba(98,0,238,1.00)",
        "borderLeftWidth": "1px",
        "borderRightColor": "rgba(98,0,238,1.00)",
        "borderRightWidth": "1px",
        "borderTopColor": "rgba(98,0,238,1.00)",
        "borderTopLeftRadius": "5px",
        "borderTopRightRadius": "5px",
        "borderTopWidth": "1px",
        "paddingBottom": "16px",
        "paddingLeft": "16px",
        "paddingRight": "16px",
        "paddingTop": "16px",
      }
    }
  >
    <canvas
      height={128}
      style={
        Object {
          "height": 128,
          "width": 128,
        }
      }
      theme={
        Object {
          "colors": Object {
            "accent": "#03dac4",
            "backdrop": "rgba(0, 0, 0, 0.5)",
            "background": "#f6f6f6",
            "disabled": "rgba(0, 0, 0, 0.26)",
            "error": "#B00020",
            "notification": "#f50057",
            "placeholder": "rgba(0, 0, 0, 0.54)",
            "primary": "#6200ee",
            "surface": "#ffffff",
            "text": "#000000",
          },
          "dark": false,
          "fonts": Object {
            "light": "Roboto-Light, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "medium": "Roboto-Medium, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "regular": "Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
            "thin": "Roboto-Thin, Roboto, \\"Helvetica Neue\\", Helvetica, Arial, sans-serif",
          },
          "roundness": 4,
        }
      }
      width={128}
    />
  </div>
</div>
`;
